lib/suppressions.cpp

```diff
@@ -43,6 +43,16 @@ static bool isAcceptedErrorIdChar(char c)
     }
 }
 
+std::size_t Hash::getHash() const
+{
+    if (mComputed)
+        return mHash;
+
+    mHash = mComputation();
+    mComputed = true;
+    return mHash;
+}
```

Comment:
We already have a [`memoize`](https://github.com/danmar/cppcheck/blob/main/lib/astutils.cpp#L1711) function that will do this for any lambda. It could be moved out of astutils.cpp into a header and reused here.